fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -677,6 +731,7 @@ private void removeEmptyFragments() {
                         throw new RepositoryRuntimeException("Error removing empty fragments", ex);
                     }
                 });
+                touch();
```

Comment:
We probably should move this `touch()` deeper into the logic (i.e. up a few lines) so that it is only called if `n.remove()` is called.